includes/class-wc-gateway-amazon-payments-advanced-privacy.php

```diff
@@ -292,14 +291,16 @@ protected function maybe_handle_order( $order ) {
 
 		$deleted = false;
 		foreach ( $meta_to_delete as $key ) {
-			$meta_value = get_post_meta( $order_id, $key, true );
+			$meta_value = $order->get_meta( $key, true, 'edit' );
 			if ( empty( $meta_value ) ) {
 				continue;
 			}
-			delete_post_meta( $order_id, $key );
+			$order->delete_meta_data( $key );
 			$deleted = true;
 		}
 
+		$order->save();
```

Comment:
Why?

Reply:
there is a foreach a couple of lines above which uses WC_Order::delete_meta_data. we need to commit the transaction with save then.

Reply:
I see, thanks for explaining!